### PR TITLE
major refactor to pure express

### PR DESCRIPTION
**Pull request by cemusta**
dropped routing controllers and typedi

### CODE REVIEW

File: src/api/app.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+app.use(cors());
+app.use(helmet());
+
+if (config.server.httpLogging) {
+  app.use(logFilter, httpLogger);
+} else {
+  logger.warn('Http logging disabled');
+}
+
+app.use('/api/Health', healthRouter);
```
Comment (lutzk) [inline src/api/app.ts:23]: i would not use upercases in the routes
also maybe we can store it in consts

File: src/api/config/index.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import * as path from 'path';
+
+class Config {
+  public auth: AuthSection;
+
+  public db: DatabaseSection;
+
+  public server: ServerSection;
+
+  constructor(envConfig: { [key: string]: any }) {
```
Comment (lutzk) [inline src/api/config/index.ts:15]: `[key: string]: any`
if its comming from dotEnv  the values are string

File: src/api/middleware/errorHandler.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  message: string;
+
+  constructor(status: number, message: string) {
+    super(message);
+    this.status = status;
+    this.message = message;
+  }
+}
+
+export const errorHandler = (error: HttpException, request: Request, response: Response, next: NextFunction) => {
```
Comment (lutzk) [inline src/api/middleware/errorHandler.ts:16]: i think there is a type for the whole handler so you would not need to type all params explicitly

File: src/controller/configController.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import type { Request, Response } from 'express';
```
Comment (lutzk) [inline src/controller/configController.ts:1]: maybe `controllers` would be better dir name

File: src/controller/configController.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+
+const getConfig = (req: Request, res: Response) => res.send({ hello: 'world' });
```
Comment (lutzk) [inline src/controller/configController.ts:3]: same as for the error handler
there is a type for `RequestHandler` from express

File: src/api/app.ts (hunk 1 above)
Comment (rafaelpinabew) [reply src/api/app.ts:23]: +1

File: src/api/middleware/httpLogger.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import * as expressPino from 'express-pino-logger';
+import type {
+  RequestHandler,
+} from 'express';
+import { logger } from '../../logger';
+
+const filteredPaths = [
+  '/api/health',
```
Comment (rafaelpinabew) [inline src/api/middleware/httpLogger.ts:8]: If we move the routes into an enum or const we could update this one to the imported one

File: src/test/utils.ts [hunk 7]
```diff
@@ -1,54 +1,54 @@
...
-  scope: 'some-scope',
-  id_token: 'some-id',
-});
-
-export {
-  createToken,
-  createJwt,
-  tokenResponse,
-};
+// import * as jwt from 'jsonwebtoken';
```
Comment (rafaelpinabew) [inline src/test/utils.ts:1]: Should we remove this file?

**Review by lutzk — APPROVED** [review]

**Pull request merged**